src/libsystemd/sd-bus/bus-message.c

```diff
@@ -4317,7 +4317,7 @@ _public_ int sd_bus_message_rewind(sd_bus_message *m, int complete) {
         return !isempty(c->signature);
 }
 
-static int message_read_ap(
+_public_ int sd_bus_message_readv(
                 sd_bus_message *m,
                 const char *types,
                 va_list ap) {
```

Comment:
Yeah, it's reasonable to export this.

Public functions cannot use `assert` on arguments, they need `assert_return` instead.

Also, to actually be able to use the function, it needs to be added to the symbols file `src/libsystemd/libsystemd.sym`.

An addition to the man page would be nice. Unfortunately there's no `sd_bus_message_read` page yet, so one would have to be written from scratch for those two functions. This is not required, since it's quite a bit of work, but if you did that, it'd be really appreciated.

Reply:
> An addition to the man page would be nice.

I plan on contributing some man page updates, as per #235 (long overdue).

I'll make the other changes momentarily. Many thanks for the prompt response; it is always appreciated.